plutus-core/executables/Common.hs

```diff
@@ -62,84 +66,205 @@ import           Text.Read                                (readMaybe)
+    printBudgetState _term model (Cek.CountingSt budget) = printBudgetStateBudget model budget
+
+instance (Eq fun, Cek.Hashable fun, Show fun) => PrintBudgetState (Cek.TallyingSt fun) where
+    printBudgetState term model (Cek.TallyingSt tally budget) = do
+        printBudgetStateBudget model budget
+        putStrLn ""
+        printBudgetStateTally term model tally
+
+instance PrintBudgetState Cek.RestrictingSt where
+    printBudgetState _term model (Cek.RestrictingSt (ExRestrictingBudget budget)) =
+        printBudgetStateBudget model budget
 
 
 ---------------- Types for commands and arguments ----------------
 
 data Input       = FileInput FilePath | StdInput
 data Output      = FileOutput FilePath | StdOutput
-data Language    = TypedPLC | UntypedPLC
 data TimingMode  = NoTiming | Timing Integer deriving (Eq)  -- Report program execution time?
 data CekModel    = Default | Unit   -- Which cost model should we use for CEK machine steps?
 data PrintMode   = Classic | Debug | Readable | ReadableDebug deriving (Show, Read)
 type ExampleName = T.Text
 data ExampleMode = ExampleSingle ExampleName | ExampleAvailable
-data EvalMode    = CK | CEK deriving (Show, Read)
 data BudgetMode  = Silent
                  | forall cost. (Eq cost, NFData cost, PrintBudgetState cost) =>
                      Verbose (Cek.ExBudgetMode cost PLC.DefaultUni PLC.DefaultFun)
-data AstNameType = Named | DeBruijn  -- Do we use Names or de Bruijn indices when (de)serialising ASTs?
+-- | @Name@ can be @Name@s or de Bruijn indices when we (de)serialise the ASTs.
+-- PLC doesn't support de Bruijn indices when (de)serialising ASTs.
+-- UPLC supports @Name@ or de Bruijn indices.
+data AstNameType =
+  Named
+  | DeBruijn
+
 type Files       = [FilePath]
 
-data Format = Plc | Cbor AstNameType | Flat AstNameType -- Input/output format for programs
+-- | Input/output format for programs
+data Format =
+  Textual
+  | Cbor AstNameType
+  | Flat AstNameType
 instance Show Format where
-    show Plc             = "plc"
+    show Textual         = "textual"
     show (Cbor Named)    = "cbor-named"
     show (Cbor DeBruijn) = "cbor-deBruijn"
     show (Flat Named)    = "flat-named"
     show (Flat DeBruijn) = "flat-deBruijn"
 
-data TypecheckOptions = TypecheckOptions Input Format
-data ConvertOptions   = ConvertOptions Language Input Format Output Format PrintMode
-data PrintOptions     = PrintOptions Language Input PrintMode
-data ExampleOptions   = ExampleOptions Language ExampleMode
-data EraseOptions     = EraseOptions Input Format Output Format PrintMode
-data EvalOptions      = EvalOptions Language Input Format EvalMode PrintMode BudgetMode TimingMode CekModel
-data ApplyOptions     = ApplyOptions Language Files Format Output Format PrintMode
+data ConvertOptions   = ConvertOptions Input Format Output Format PrintMode
+data PrintOptions     = PrintOptions Input PrintMode
+newtype ExampleOptions   = ExampleOptions ExampleMode
```

Comment:
I'm afraid I can't stop myself: I'm going to be obsessive about the alignment here.  There's perhaps no compelling reason for `ExampleOptions` to be a newtype since we don't care about efficiency here, and changing it back to `data` would make everything line up again.

Reply:
Sorry it's me applying the linter suggestions without much thinking to make them go away.